src/sid/virus_strains.py

```diff
@@ -0,0 +1,102 @@
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+from typing import Union
+
+import numpy as np
+import pandas as pd
+from sid.config import DTYPE_VIRUS_STRAIN
+
+
+def combine_first_factorized_infections(
+    first: Union[pd.Series, np.ndarray], second: Union[pd.Series, np.ndarray]
+) -> Union[pd.Series, np.ndarray]:
+    """Combine factorized infections where the first has precedence."""
+    combined = second.copy()
+    combined[first >= 0] = first[first >= 0]
+    return combined
+
+
+def categorize_factorized_infections(
+    factorized_infections: Union[pd.Series, np.ndarray], virus_strains: Dict[str, Any]
+) -> pd.Series:
+    """Convert factorized infections with virus strains to a categorical."""
+    return pd.Series(
+        pd.Categorical(
+            factorized_infections, categories=range(-1, len(virus_strains["names"]))
+        )
+        .rename_categories(["not_infected"] + virus_strains["names"])
+        .remove_categories("not_infected")
+    )
+
+
+def factorize_multiple_boolean_or_categorical_infections(
```

Comment:
If I see correctly, this is only used for initial infections. Thus, I would choose a less abstract name like `factorize_initial_infections`. It took me a while to understand what is happening in the function and having a less abstract name would have helped.